### PR TITLE
adapt to windows (msvc) and use vcpkg for 3rd libs (include curlpp)

### PR DESCRIPTION
**Pull request by kkzi**
Recommended to use [vcpkg ](microsoft/vcpkg)  as a 3rd lib manager.

1. adapt to windows msvc (2019+):  add src/win/[getopt.h](skandhurkat/Getopt-for-Visual-Studio) 
2. use vcpkg instead of traditional find_package
3. remove curlpp submodule, also use vcpkg

### CODE REVIEW

**Comment by harshavardhana** [comment]
Please fix the CI builds @kkzi

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -16,7 +16,7 @@
 SET(SRCS s3.cpp s3_http.cpp s3_io.cpp s3_signature_v2.cpp)
 SET(AWS_INSTALL_LIST)
 
-add_library(miniocpp SHARED ${SRCS})
+add_library(miniocpp STATIC ${SRCS})
```
Comment (harshavardhana) [inline src/CMakeLists.txt:19]: why is this being changed? @kkzi

File: examples/win/getopt.h [hunk 2]
```diff
@@ -0,0 +1,656 @@
+#ifndef __GETOPT_H__
+/**
+ * DISCLAIMER
+ * This file is part of the mingw-w64 runtime package.
+ *
+ * The mingw-w64 runtime package and its code is distributed in the hope that it 
+ * will be useful but WITHOUT ANY WARRANTY.  ALL WARRANTIES, EXPRESSED OR 
+ * IMPLIED ARE HEREBY DISCLAIMED.  This includes but is not limited to
```
Comment (harshavardhana) [inline examples/win/getopt.h:8]: why is this added to this repo? @kkzi

File: .gitignore [hunk 3]
```diff
@@ -6,4 +6,6 @@ libminiocpp.so
 *.a
 *.so
 build/
-*.pc
\ No newline at end of file
+*.pc
+.vs
+CMakeSettings.json
```
Comment (harshavardhana) [inline .gitignore:11]: where is CMakeSettings.json? and why should it be ignored?

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Please incorporate https://github.com/marketplace/actions/run-vcpkg and add `vcpkg.json` to make sure that this project builds.

File: src/CMakeLists.txt (hunk 1 above)
Comment (kkzi) [reply src/CMakeLists.txt:19]: there are no export symbols in any header files, which cause link errors on windows with msvc.
therefore, it is more appropriate to use a static link library, for both gcc and msvc. 
define export symbols in header file is also a solution.
the most suitable solution is to support both dynamic linking and static linking, it will take some hours and need planning

File: examples/win/getopt.h (hunk 2 above)
Comment (kkzi) [reply examples/win/getopt.h:8]: to support windows with msvc. 
there is  no <getopt.h> in msvc toolchain.

File: .gitignore (hunk 3 above)
Comment (kkzi) [reply .gitignore:11]: i use visual studio 2019 (msvc), will generate these directories and files

File: src/CMakeLists.txt (hunk 1 above)
Comment (harshavardhana) [reply src/CMakeLists.txt:19]: Understood, but this is a breaking change I don't think I can accept it right now. It is okay for this project to not build on windows. As long as it builds on Linux for now we are good.

File: examples/win/getopt.h (hunk 2 above)
Comment (harshavardhana) [reply examples/win/getopt.h:8]: In that case it should move to `include/win/getopt.h`

File: .gitignore (hunk 3 above)
Comment (harshavardhana) [reply .gitignore:11]: Okay understood.

File: src/CMakeLists.txt (hunk 1 above)
Comment (harshavardhana) [reply src/CMakeLists.txt:19]: Actually chuck that, we are good here.

File: examples/win/getopt.h (hunk 2 above)
Comment (harshavardhana) [reply examples/win/getopt.h:8]: Actually not needed since this is only used by examples for now.

**Review by harshavardhana — APPROVED** [review]
